src/backend/distributed/utils/resource_lock.c

```diff
@@ -316,3 +318,69 @@ LockRelationShardResources(List *relationShardList, LOCKMODE lockMode)
 		}
 	}
 }
+
+
+/*
+ * LockParentShardResourceIfPartition checks whether given shard belongs to a
```

Comment:
Pretty minor comments: 
`whether given shard` -> `whether the given shard `

`If it is, ` -> `If it does` ?